common/util.go

```diff
@@ -434,14 +433,16 @@ func CreateHistoryStartWorkflowRequest(
 		StartRequest: startRequest,
 	}
 
-	firstDecisionTaskBackoffSeconds := backoff.GetBackoffForNextScheduleInSeconds(
-		startRequest.GetCronSchedule(), now, now)
 	delayStartSeconds := startRequest.GetDelayStartSeconds()
-	if delayStartSeconds > 0 && firstDecisionTaskBackoffSeconds > 0 {
-		return nil, ErrDelayStartSeconds
-	}
-	if delayStartSeconds > 0 {
-		firstDecisionTaskBackoffSeconds = delayStartSeconds
+	firstDecisionTaskBackoffSeconds := delayStartSeconds
+	if len(startRequest.GetCronSchedule()) > 0 {
+		delayedStartTime := now.Add(time.Second * time.Duration(delayStartSeconds))
+		firstDecisionTaskBackoffSeconds = backoff.GetBackoffForNextScheduleInSeconds(
+			startRequest.GetCronSchedule(), delayedStartTime, delayedStartTime)
+
+		// backoff seconds was calculated based on delayed start time, so we need to
+		// add the delayStartSeconds to that backoff.
+		firstDecisionTaskBackoffSeconds += delayStartSeconds
```

Comment:
My understanding is the cron is calculated based on the now+delay time. Then we don't need to add this back?

Reply:
I think we need to define the delay + cron behavior is:
1. Delay then calculate cron 
2. Calculate the cron based on delay start time

Reply:
> My understanding is the cron is calculated based on the now+delay time. Then we don't need to add this back?

I tried to put a comment to explain this but let me know how I should update this. Imagine we want to fire the cron scheduler every 10 seconds, want to delay 3 seconds and we are at 0th second.

- backoffSeconds are calculated based on 0+3 seconds so it returns 7 seconds
- we still need to add 3 seconds so we can start at the right time. 

Let me know if this makes sense.

Reply:
I see. It makes sense to me now.